### ISSUE TITLE
OACMPDM10 water sound bug

### ISSUE THREAD

**Issue by EddieBrrrock** [NeonKnightOA/oacmpvol1#39]
water sound stop immediately when walking between this two pillars :
![shot0026](https://user-images.githubusercontent.com/99912467/162945731-fa57cfa7-35ac-419d-a795-750a8f160b2f.jpg)
Or behind this pillar :
![shot0027](https://user-images.githubusercontent.com/99912467/162945767-87cb43ab-eb71-4566-8d41-fa15c4aa079d.jpg)

**Comment by EddieBrrrock** [comment]
z_oacmp1v4_beta-220411

**Comment by EddieBrrrock** [comment]
Edit : others pillars like this have this problem

**Comment by EddieBrrrock** [comment]
Yes i'm using openal and disabling it remove the problem but the sound it still quite strange around theses pillars
![Capture d’écran 2022-04-12 à 19 31 59](https://user-images.githubusercontent.com/99912467/163020725-bd1b9b7a-9d26-43ba-8a20-8ddc4da48d76.png)

**Comment by EddieBrrrock** [comment]
By the way, isn't there supposed to be a waterfall here?
![shot0039](https://user-images.githubusercontent.com/99912467/163021075-016055f4-f1ce-400a-8ba3-c8c2f656987f.jpg)

**Comment by The-Gig** [comment]
I've done a few tests, recompiling the map with different settings. It looks like that OpenAL mode uses VIS data: when you move yourself so the area where the target_speaker is located isn't rendered anymore, that sound stops abruptly.
I can see a rationale behind this.... something like "if it's not rendered, it's because you cannot see it, so _probably_ it's another room, with a wall in the middle which _may_ block the sound in real life". But in practice it doesn't apply well to any situation, as VIS could just be blocked by a column and in real life you don't abruptly stop hearing a sound as long as you exit the door of the room.
I don't know if this applies only to target speaker or to all in-game sounds, like footsteps and weapon shots...

For completeness, here it is the infos about the OpenAL driver I installed (the original Creative Labs one, on Windows 10)
![immagine](https://user-images.githubusercontent.com/23258514/165738933-e21c8c68-2224-41f7-b78c-7a52c946b74d.png)

Anyway, I don't know if it should be considered a _map_ bug or an _engine_ bug (maybe something like adding s_useopenal 2 to have openal but ignoring VIS? IDK).... and I'm unsure about how to workaround it.
Suggestions about how to "fix"? 
- The map is already relatively heavy, the idea of compiling it without any VIS optimization doesn't really excite me (although most of the map is already rendered from most places)...
- Trying making the bridge (and the cistern) "detail" instead of "structural", giving up only _part_ of the little VIS optimization it has got?
- Moving the target_speaker entities, adding some more of them (with wav files at lower volume) to try better covering the area?

**Comment by The-Gig** [comment]
Just for the record, I tried the map in Q3 with ioquake3, and the same happens with OpenAL active.

**Comment by The-Gig** [comment]
I opened a github issue in the OA engine repo here: https://github.com/OpenArena/engine/issues/78

**Comment by NeonKnightOA** [comment]
So this doesn't seem to be related to the map itself, isn't it?

**Comment by The-Gig** [comment]
Currently trying the method of making lower volume sounds for the water, and applying more of them (placing the new ones under the bridge), and that seems to me to have made the situation better. I hope to be able to give you a playable beta tomorrow.

**Comment by The-Gig** [comment]
Applying the workaround took me more time than expected, I'll give you a beta in the next week as I will not be able to work on it for a couple of days.

**Comment by The-Gig** [comment]
Hi @Bishop-333 
better late than never, here's an udestruction update (to be launched as "udestruction" instead of "oacmpdm10" for more easily compare with oacmpvol1v3 version) with a workaround for the OpenAL bug (although not perfect, the issue is much less noticeable now, I think) and some other fixes under the hood.
[z_udestruction_version30.pk3.zip](https://github.com/NeonKnightOA/oacmpvol1/files/15078681/z_udestruction_version30.pk3.zip)
What do you think?

Note: probably the version of this beta should have been 21, but I wasn't sure so I took a leap and named it "version 30".
